crates/db-device/README.md

```diff
@@ -0,0 +1,8 @@
+# db-device
+
+db-device crate is designed to support the virtual machine's device model, which mainly contians:
+
+- `DeviceIo` and `DeviceIoMut` trait: defining the read/write operation on specificed device
+- `IoManager`: managing all devices of virtual machine, register their io resource callback
+- `IoManagerContext` trait: transaction context for `IoManager` to (un)register devices
+- `ResourceConstraint`, `Resource` and `DeviceResources`: abstractions for requiring and describing resources of a device
```

Comment:
Could you have a simple example/usage part here to show how to use the crate and API interaction?

Reply:
added